template/src/features/counter/Counter.js

```diff
@@ -42,14 +43,20 @@ export function Counter() {
         <button
           className={styles.button}
           onClick={() =>
-            dispatch(
-              incrementByAmount({ amount: Number(incrementAmount) || 0 })
-            )
+            dispatch(incrementByAmount(Number(incrementAmount) || 0))
           }
         >
           Add Amount
         </button>
       </div>
+      <div className={styles.row}>
```

Comment:
The layout here feels a bit confusing to me after looking at the gif, in a couple ways:

- The buttons aren't aligned
- It's surprising to me that "Add Async" doesn't add the amount in the textbox

Can we put the textbox and both buttons in a single row, and make this do `dispatch(incrementAsync(incrementAmount)` ?  That also lets us show passing an argument to a thunk while we're at it.

Reply:
Actually, given that, would it be better to store the amount as a number permanently, and do the number conversion in the `onChange` handler?

Reply:
I think it will lead to a simpler example if we leave it stored as a string. Storing as a number means we'd be doing a form of input masking, which requires more code to make work properly for things like `""`, `"1."`, `"001"`, and other intermediate forms of number inputs. I talked a little more about this in https://github.com/reduxjs/cra-template-redux-typescript/pull/3#discussion_r382957207